tests/access_test.go

```diff
@@ -53,6 +54,57 @@ var _ = Describe("[rfe_id:500][crit:high][vendor:cnv-qe@redhat.com][level:compon
 		tests.BeforeTestCleanup()
 	})
 
+	Describe("With the kubevirt-controller service account", func() {
+		table.DescribeTable("should verify permissions on resources are correct for kubevirt-controller", func(resource string) {
+
+			controllerVerbs := make(map[string]string)
+			controllerVerbs["list"] = "yes"
+			controllerVerbs["watch"] = "yes"
+			controllerVerbs["delete"] = "no"
+			controllerVerbs["update"] = "no"
+			controllerVerbs["deleteCollection"] = "no"
+
+			// Adjustment for the rule associated with kubevirt, vmipresets, vmim
+			switch resource {
+			case "kubevirt":
+				controllerVerbs["get"] = "yes"
+				controllerVerbs["patch"] = "no"
+				controllerVerbs["create"] = "no"
+			case "virtualmachineinstancepresets":
+				controllerVerbs["get"] = "no"
+				controllerVerbs["patch"] = "no"
+				controllerVerbs["create"] = "no"
+			case "virtualmachineinstancemigrations":
+				controllerVerbs["get"] = "yes"
+				controllerVerbs["patch"] = "yes"
+				controllerVerbs["create"] = "yes"
+			default:
+				controllerVerbs["get"] = "yes"
+				controllerVerbs["patch"] = "yes"
+				controllerVerbs["create"] = "no"
+			}
+
+			namespace := tests.NamespaceKubevirt // Is there an exported var defines kubevirt ns ?
+			verbs := []string{"get", "list", "watch", "delete", "create", "update", "patch", "deletecollection"}
+
+			for _, verb := range verbs {
+				By(fmt.Sprintf("verifying kubevirt-controller for verb %s", verb))
+				expectedRes, _ := controllerVerbs[verb]
+				as := fmt.Sprintf("system:serviceaccount:%s:%s", namespace, controller)
+				result, _, _ := tests.RunCommand(k8sClient, "auth", "can-i", "--as", as, verb, resource)
```

Comment:
lets use the virtClient for this instead 
```
virtClient.AuthorizationV1().SelfSubjectAccessReviews().Create(v1.SelfSubjectAccessReview{
  ResourceAttributes: &v1.ResourceAttributes{}
})
```

Reply:
I found the use and def of `virtClient` in `vm` related functests. They are all defined as 'kubecli.KubevirtClient`. I am guess it's not come out of 'service account'. 

The object `virtClient.AuthorizationV1().SelfSubjectAccessReviews()` in your reference were from `staging/src`, no usage under `tests/`. 

According what was stated in the issue #3512 , which was to limit the permission of service account `kubevirt-controller` from deleting/altering the custom resources in the `kubevirt.io` API group. 

So the RBAC rule modification was for the ClusterRole `kubevirt-controller`, and the object of the test was the service account `kubevirt-controller`, in the namespace `kubevirt`. 

Do we know if the `kubecli` binds to the `kubevirt-controller` ClusterRole? From a deployed cluster in my dev environment, I do not see other binding that binds to this ClusterRole. I do not know about the CI environment.

Reply:
I think Ashley meant `SubjectAccessReviews` and then you can fill `SubjectAccessReview` with `user`, `verb`...
If you were using a `SelfSubjectAccessReviews` then `user` would be derived from `kubeconfig` or `token`, which would not be `kubevirt-controller` as you pointed out. 

The difference between your approach vs using `virtClient` is that `virtClient` is part of compiled code and tests might be faster(?).

Reply:
Thank you! yes  I meant `SubjectAccessReviews`. My thought was more so that if you don't need to shell out to run a command you shouldn't. But it also looks like that is what the other tests in this file are doing so maybe it doesn't really matter.